autosubmit/git/autosubmit_git.py

```diff
@@ -45,6 +42,45 @@
     )''', re.VERBOSE)
 """Regular expression to match Git URL."""
 
+
+# M: Modified, A: Added, D: Deleted, ?: Untracked
+_GIT_STATUS_DIRTY_FLAG = ["M", "A", "D", "?"]
+"""List of git statuses. The output of ``git status --porcelain`` contain
+flags to indicate the status of the object."""
+
+
+def check_unpushed_changes(expid: str, as_conf: AutosubmitConfig) -> None:
+    """Check if the Git repository is dirty for an operational experiment.
+
+    Raises an AutosubmitCritical error if the experiment is operational,
+    the platform is Git, and there are unpushed changes in the local Git
+    repository, or in any of its Git submodules.
+
+    :param expid: The experiment ID.
+    :param as_conf: Autosubmit configuration object.
+    """
+    project_type = as_conf.get_project_type()
+    if expid[0] == 'o' and project_type == 'git':
+        git_cmds = [
+            ["git"],
+            ["git", "submodule"]
+        ]
+        arguments = [
+            ["status", "--porcelain"],
+            ["foreach", "'git status --porcelain'"]
+        ]
+
+        proj_dir = as_conf.get_project_dir()
+
+        for git_cmd, args in zip(git_cmds, arguments):
+            output = subprocess.check_output([*git_cmd, *args], cwd=proj_dir)
+            output = output.decode(locale.getlocale()[1])
+            for line in output.splitlines():
+                tokens = line.strip().split(' ')
+                if tokens[0] in _GIT_STATUS_DIRTY_FLAG:
+                    raise AutosubmitCritical("Push Git local changes to remote repository before running", 7075)
+
```

Comment:
The function was already static, and it had no state-dependency to the class. So it made no difference the function being here in the module, or in an object. The advantage for this one is that we don't need to create an object, avoid risks of state, and it could be faster (I think it's faster to load from a module directly?).

But also simpler to test/call.